src/db_cache/mod.rs

```diff
@@ -404,6 +404,60 @@ pub mod stats {
     }
 }
 
+#[cfg(test)]
+pub(crate) mod test_utils {
```

Comment:
this is just moved from the `test` module so other parts of the codebase can access a simple in-memory cache for testing